cpp/models/ide_secir/parameters_io.cpp

```diff
@@ -0,0 +1,231 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "ide_secir/parameters_io.h"
+#include "memilio/config.h"
+
+#ifdef MEMILIO_HAS_JSONCPP
+
+#include "ide_secir/model.h"
+#include "ide_secir/infection_state.h"
+#include "memilio/math/eigen.h"
+#include "memilio/io/epi_data.h"
+#include "memilio/io/io.h"
+#include "memilio/utils/date.h"
+
+#include <string>
+#include <cmath>
+
+namespace mio
+{
+namespace isecir
+{
+
+// We assume that we start the simulation at time 0 and want to compute the necessary flows
+// in the past for the initialization of an IDE-SECIR model based on real data.
+IOResult<void> set_initial_flows(Model& model, ScalarType dt, std::string const& path, Date date,
+                                 ScalarType scale_confirmed_cases)
+{
+    //--- Preparations ---
+    // Try to get rki data from path.
+    BOOST_OUTCOME_TRY(rki_data, mio::read_confirmed_cases_noage(path));
+    auto max_date_entry = std::max_element(rki_data.begin(), rki_data.end(), [](auto&& a, auto&& b) {
+        return a.date < b.date;
+    });
+    if (max_date_entry == rki_data.end()) {
+        log_error("RKI data file is empty.");
+        return failure(StatusCode::InvalidFileFormat, path + ", file is empty.");
+    }
+    auto max_date = max_date_entry->date;
+    if (max_date < date) {
+        log_error("Specified date does not exist in RKI data.");
+        return failure(StatusCode::OutOfRange, path + ", specified date does not exist in RKI data.");
+    }
+
+    // Get (global) support_max to determine how many flows in the past we have to compute.
+    ScalarType global_support_max         = model.get_global_support_max(dt);
+    Eigen::Index global_support_max_index = Eigen::Index(std::ceil(global_support_max / dt));
+
+    // m_transitions should be empty at the beginning.
+    if (model.m_transitions.get_num_time_points() > 0) {
+        model.m_transitions = TimeSeries<ScalarType>((int)InfectionTransition::Count);
+    }
+
+    // The first time we need is -4 * global_support_max.
+    Eigen::Index start_shift = -4 * global_support_max_index;
+    // The last time needed is dependent on the mean stay times in the Exposed compartment and in InfectedNoSymptoms
+    // before transition to InfectedSymptoms.
+    Eigen::Index last_time_index_needed = Eigen::Index(std::ceil(
+        (model.parameters.get<TransitionDistributions>()[Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms)]
+             .get_mean(dt) +
+         model.parameters
+             .get<TransitionDistributions>()[Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms)]
+             .get_mean(dt)) /
+        dt));
```

Comment:
Below you are defining `mean_InfectedNoSymptomsToInfectedSymptoms`. You could already do this here, I think this would make the connection clearer.

Reply:
Good idea